tools/ctb-tile.cpp

```diff
@@ -451,6 +690,14 @@ runTiler(TerrainBuild *command, Grid *grid) {
 
   GDALClose(poDataset);
 
+  // Pass metadata to global instance.
+  if (threadMetadata) {
+    static std::mutex mutex;
```

Comment:
I could be wrong, but this seems broken. We're creating a new mutex instance in every thread, rather than having all threads share the same global mutex.

Reply:
But, it is declared like static, I think it is shared by all threads

Reply:
Am I wrong?

Reply:
Ah, that might work.  Sorry, my C++ is pretty shaky :)

I'm actually trying to use your branch as a starting point for implementing MBTiles support in CTB, so I'm having to dust off my C++ skills for the first time in years.

Reply:
Don't worry. You are welcome! umm MBTiles support! it looks great @markerikson